File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDiffInBST(self, root: Optional[TreeNode]) -> int:
        
        arr=[]
        def inorder(root):
            
            if not(root):
                return 
            
            
            
            inorder(root.left)
            arr.append(root.val)
            inorder(root.right)
        
        inorder(root)
        res=arr[len(arr)-1]
        for i in range(len(arr)-1):
              
            diff=arr[i+1]-arr[i]
            res=min(res,diff)
            
        return res        
                
        

File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = function(root) {
    
    let res=Infinity;
    
    let prev=null;
    
    
    function inorder(root){
        
        if(!root){
            
            return
        }
        
        inorder(root.left)
        
        if (prev){
            
            res=Math.min(res,root.val-prev.val)
        }
        prev=root;
        inorder(root.right)
    }
    inorder(root)
    return res
    
    
    
};